src/Swarm/Game/CEK.hs

```diff
@@ -253,3 +262,31 @@ prettyFrame FLoadEnv {} = "loadEnv"
 prettyFrame FExec = "exec _"
 prettyFrame (FBind Nothing t _) = "_ ; " ++ prettyString t
 prettyFrame (FBind (Just x) t _) = from x ++ " <- _ ; " ++ prettyString t
+prettyFrame FImmediate {} = "(_ : cmd a)"
+
+--------------------------------------------------------------
+-- Wrappers for functions in FImmediate
+--
+-- NOTE: we can not use GameState and Robot directly, as it
+-- would create a cyclic dependency. The alternative is
+-- making CEK, Cont and Frame polymorphic which just muddies
+-- the picture too much for one little game feature.
+--
+-- BEWARE: the types do not follow normal laws for Show and Eq
+--------------------------------------------------------------
+
+newtype WorldFun = WorldFun
```

Comment:
I'm pretty sure we already have something called `WorldFun` in `Swarm.Game.World`.  Perhaps we should call this something else.  `WorldUpdate`, perhaps?  We could also change the name of `RobotFun` to match.

Reply:
Somehow I forgot about `WorldFun`, now it is fixed.